internal/controllers/objecttemplate/template_reconciler.go

```diff
@@ -94,14 +95,23 @@ func (r *templateReconciler) Reconcile(
 		return res, err
 	}
 
-	if err := r.dynamicCache.Watch(
-		ctx, objectTemplate.ClientObject(), obj); err != nil {
-		return res, fmt.Errorf("watching new child: %w", err)
+	objects := []client.Object{
+		objectTemplate.ClientObject(),
+		obj,
+	}
+	cache, err := r.accessManager.GetWithUser(
+		context.Background(),
```

Comment:
Can we use the existing context here?
```suggestion
		ctx,
```